src/index.ts

```diff
@@ -121,4 +159,87 @@ const s3plugin: JupyterFrontEndPlugin<IS3Auth> = {
   }
 };
 
-export default [s3plugin, apiClientPlugin];
+/**
+ * The project information provider.
+ */
+const projectInfo: JupyterFrontEndPlugin<IProjectInfo> = {
+  id: 'jupybase:project-info-provider',
+  description: 'The project information provider.',
+  requires: [IJupybaseClient],
+  provides: IProjectInfo,
+  activate: async (
+    app: JupyterFrontEnd,
+    jupybaseClient: IJupybaseClient
+  ): Promise<IProjectInfo> => {
+    // extract the project id from the url
+    const searchParams = new URLSearchParams(window.location.search);
+    const projectId = searchParams.get('id');
+
+    // get the project information
+    const apiClient = jupybaseClient.apiClient;
+    const url = getURL(`/api/v1/env/${projectId}`);
+
+    const result = await apiClient?.get(url);
+    const content = result!.data.data[0].content;
```

Comment:
you should wrap this call in a try-catch block, for example

```suggestion
    let content = {name: '', shortDesc : '',  buildEnv : '',  kernelEnv : '', lockfile: '', dependencies : [] }
    try{
        const result = await apiClient?.get(url);
        content = result!.data.data[0].content;
    } catch {
     // handle error here
    }

```